views/request_product_views.py

```diff
@@ -86,7 +86,8 @@ def perform_create(self, serializer):
                 template = f'{request_product.name} was requested',
                 category = request_product.category,
                 has_custom_users_target = True,
-                persons = list(persons_to_be_notified)
+                persons = list(persons_to_be_notified),
+                send_only_to_subscribed_users = True
```

Comment:
I don't think  ```send_only_to_subscribed_users``` is needed here.
Only those who have added the product earlier will be informed.
@MihirSachdeva

Reply:
Not using `send_only_to_subscribed_users` is causing this issue: A user who has requested for/added for sale an item of a particular category is not able to unsubscribe from notifications of that category.

A user should be able to manage his notification subscriptions through the notification settings.